cmd/agentctl/cli/cli.go

```diff
@@ -163,38 +166,100 @@ func (cli *AgentCli) Initialize(opts *ClientOptions, ops ...InitializeOpt) error
 			return err
 		}
 	}
+
 	if opts.Debug {
 		debug.Enable()
 		SetLogLevel("debug")
 	} else {
 		SetLogLevel(opts.LogLevel)
 	}
 
+	ReadConfig() // TODO: maybe move it elsewhere
+
 	if cli.client == nil {
-		cli.client, err = newAPIClient(opts)
+		clientOptions := buildClientOptions()
+		cli.client, err = client.NewClientWithOpts(clientOptions...)
 		if err != nil {
 			return err
 		}
 	}
 	cli.clientInfo = ClientInfo{
-		DefaultVersion: cli.client.ClientVersion(),
+		DefaultVersion: cli.client.Version(),
 	}
 	cli.initializeFromClient()
 	return nil
 }
 
-func newAPIClient(opts *ClientOptions) (client.APIClient, error) {
+func buildClientOptions() []client.Opt {
+	logging.Debug("----------------------------------------------------")
+	logging.Debug("Building client options")
```

Comment:
It would be easier and more explicit to use [`viper.Debug()`](https://godoc.org/github.com/spf13/viper#Debug) or [`viper.AllSettings()`](https://godoc.org/github.com/spf13/viper#AllSettings) for debugging actual config contents.

Reply:
Hpw about this?
```
	ReadConfig() // TODO: maybe move it elsewhere
	if opts.Debug {
		bConfig, err := json.MarshalIndent(viper.AllSettings(), "", " ")
		if err != nil {
			return fmt.Errorf("error while marshaling config to json: %v", err)
		}
		logging.Debugf("Current config: \n%s\n", string(bConfig))
	}
```